core/src/main/java/tachyon/master/TachyonMaster.java

```diff
@@ -189,6 +189,11 @@ private boolean isFormatted(String folder, String path) throws IOException {
       folder += TachyonURI.SEPARATOR;
     }
     UnderFileSystem ufs = UnderFileSystem.get(folder, mTachyonConf);
+
+    if (UnderFileSystem.isDummyUnderFS(mTachyonConf)) {
```

Comment:
Would it make more sense to perhaps move the logic of this function to the `UnderFileSystem` class then individual implementations can determine whether they are formatted or not?